LoanCalc.java

```diff
@@ -1,54 +1,74 @@
-	}
+    public static void main(String[] args) {		
+        // Gets the loan data
+        double loanAmount = Double.parseDouble(args[0]);
+        double interestRate = Double.parseDouble(args[1]);
+        int numberOfPayments = Integer.parseInt(args[2]);
+        System.out.println("Loan = " + loanAmount + ", interest rate = " + interestRate + "%, periods = " + numberOfPayments);
+        // Computes the periodical payment using brute force search
+        System.out.print("\nPeriodical payment, using brute force: ");
+        System.out.println((int) bruteForceSolver(loanAmount, interestRate, numberOfPayments, epsilon));
+        System.out.println("number of iterations: " + iterationCounter);
 
-	// Computes the ending balance of a loan, given the loan amount, the periodical
-	// interest rate (as a percentage), the number of periods (n), and the periodical payment.
-	private static double endBalance(double loan, double rate, int n, double payment) {	
-		// Replace the following statement with your code
-		return 0;
-	}
-	
-	// Uses sequential search to compute an approximation of the periodical payment
-	// that will bring the ending balance of a loan close to 0.
-	// Given: the sum of the loan, the periodical interest rate (as a percentage),
-	// the number of periods (n), and epsilon, the approximation's accuracy
-	// Side effect: modifies the class variable iterationCounter.
-    public static double bruteForceSolver(double loan, double rate, int n, double epsilon) {
-		// Replace the following statement with your code
-		return 0;
+        // Computes the periodical payment using bisection search
+        System.out.print("\nPeriodical payment, using bi-section search: ");
+        System.out.println((int) bisectionSolver(loanAmount, interestRate, numberOfPayments, epsilon));
+        System.out.println("number of iterations: " + iterationCounter);
+    }
+    // Computes the ending balance of a loan, given the loan amount, the periodical
+    // interest rate (as a percentage), the number of periods (n), and the periodical payment.
+    private static double endBalance(double loanAmount, double interestRate, int numberOfPeriods, double payment) {	
+        for (int i = 0; i < numberOfPeriods; i++) {
+            loanAmount = loanAmount - payment;
+            loanAmount = loanAmount * (1 + (interestRate / 100));
+        }
+        return loanAmount;
+    }
+    
+    // Uses sequential search to compute an approximation of the periodical payment
+    // that will bring the ending balance of a loan close to 0.
+    // Given: the sum of the loan, the periodical interest rate (as a percentage),
+    // the number of periods (n), and epsilon, the approximation's accuracy
+    // Side effect: modifies the class variable iterationCounter.
+    public static double bruteForceSolver(double loanAmount, double interestRate, int numberOfPeriods, double epsilon) {
+        double guess;
```

Comment:
No need to declare the variable if you are assigning a value to it the next row :)